segmentation/utils/visualize.py

```diff
@@ -18,3 +18,84 @@ def visualize_z_scores(z, pheno_titles):
     zplot = pd.DataFrame(z, columns=pheno_titles, index=pheno_titles)
     sns.set(font_scale=.7)
     sns.clustermap(zplot, figsize=(8, 8), cmap="vlag")
+    
+def newDict(names):
+    result = {}
+    for name in names:
+        result[name] = [0]
+    return result
+
+def visualize_distribution_of_cell_count(df, id_col_name, cell_col_name):
+    ids = set(df[id_col_name])
+    print(ids)
+    cell_names = df[cell_col_name].value_counts().index
+    cell_names = cell_names.tolist()  
+
+    unique_ids = []
+    for item in ids:
+        if type(item) is float or type(item) is int:
+            unique_ids.append(item)
+
+    unique_ids = [x for x in unique_ids if x == x]
+
+    print(unique_ids)
+
+    df_stacked = pd.DataFrame(newDict(cell_names))
+
+    first = True
+    for id in unique_ids:
+
+        df_rows = df.loc[df["PatientID"] == id]
+        row_cell_types = df_rows["cell_type"].value_counts()
+
+        if first:
+            df_stacked = pd.DataFrame([row_cell_types.values], columns=row_cell_types.index)
+            first = False
+        else:
+            df_stacked = pd.concat([df_stacked, (pd.DataFrame([row_cell_types.values], columns=row_cell_types.index))], axis=0, ignore_index=True)
+
+    df_stacked.plot.bar(stacked=True)
+    plt.legend(loc='center left', bbox_to_anchor=(1.0, 0.5))
+    plt.xlabel("Patient ID")
+    plt.ylabel("Cell Count")
+    plt.title("Distribution of Cell Count in Patients")
+
+def visualize_proportion_of_cell_count(df, id_col_name, cell_col_name):
+    ids = set(df[id_col_name])
+    cell_names = df[cell_col_name].value_counts().index
+    cell_names = cell_names.tolist()  
+    unique_ids = []
+
+    for item in ids:
+        if type(item) is float or type(item) is int:
+            unique_ids.append(item)
+
+    unique_ids = [x for x in unique_ids if x == x]
+
+    df_stacked = pd.DataFrame(newDict(cell_names))
+
+    first = True
+    for id in unique_ids:
+
+        df_rows = df.loc[df["PatientID"] == id]
+        row_cell_types = df_rows["cell_type"].value_counts(normalize=True)*100
+
+        if first:
+             df_stacked = pd.DataFrame([row_cell_types.values], columns=row_cell_types.index)
+            first = False
+        else:
+            df_stacked = pd.concat([df_stacked, (pd.DataFrame([row_cell_types.values], columns=row_cell_types.index))], axis=0, ignore_index=True)
+
+    df_stacked.plot.bar(stacked=True, legend = False)
+    plt.legend(loc='center left', bbox_to_anchor=(1.0, 0.5))
+    plt.xlabel("Patient ID")
+    plt.ylabel("Cell Proportion")
+    plt.title("Distribution of Cell Count Proportion in Patients")
```

Comment:
This function looks very similar to visualize_distribution_of_cell_count. Unless you expect the functionality of these functions to change drastically over time, I'd recommend combining much of the shared logic into one function.

Reply:
We'll combine after we make the improvements to our plotting functions